docs/RFCS/raft_consistency_checker.md

```diff
@@ -0,0 +1,86 @@
+- Feature Name: Raft consistency checker
+- Status: draft
+- Start Date: 2016-02-10
+- Authors: Ben Darnell, Bram Grunier, Vivek Menezes
+- RFC PR: (PR # after acceptance of initial draft)
+- Cockroach Issue: #837
+
+# Summary
+
+An online system that checks that periodically checks that range replicas
+are consistent; a snapshot for a range at a specific point in the raft log
+ has the same sha on all replicas.
+
+# Motivation
+
+Consistency! Correctness at scale.
+
+# Detailed design
+
+The proposal is for each range replica leader to periodically run the
+consistency checker. The consistency checker runs in three phases:
+
+1. All replicas agree on the snapshot at which a sha is to be computed.
+2. A sha is computed on all replicas.
+3. The leader shares its sha with all replicas; a replica computing a
+different sha from the one supplied by the leader alerts and eutanizes
+the node.
+
+The above is implemented by introducing two raft commands:
+
+1. CHECK_CONSISTENCY_BEGIN (BEGIN): A replica on receiving this command
+immediately acknowledges it has executed it, and asynchronously computes the sha
+for the current snapshot. It continues executing commands while it computes the
+sha on the side. Once the sha is computed it compares the BEGIN command id along
```

Comment:
There's a bit missing here about creating a uuid for the request.  That's the payload that BEGIN sends.  Command ID should be replaced by that uuid.
We want to make sure the id is unique since any collision could result in a failure of consistency check.